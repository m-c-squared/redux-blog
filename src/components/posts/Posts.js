import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';
import styles from 'styled-components';

const PostLists = styles.ul`
@media (min-width: 374px) {
  padding: 10px; 
  }
@media (min-width: 700px) {
  display: grid;
  grid-template-columns: 33% 33% 33%; 
  grid-column-gap: 30px;
  list-style: none;
}
`;

const PostItems = styles(Link)`
  @media (min-width: 374px) {
    font-family: 'Roboto', san-serif;
    color: rgb(93, 95, 96);
  }
  @media (min-width: 700px) {
    font-family: 'Roboto', san-serif;
    color: rgb(93, 95, 96);
    font-size: 30px;
  }

`;


const PostDes = styles.p`
  @media (min-width: 374px) {
    font-family: 'Roboto', san-serif;
    color: rgb(93, 95, 96);
  }
  @media (min-width: 700px) {
    font-family: 'Roboto', san-serif;
    color: rgb(93, 95, 96);
    font-size: 25px;
  }
`;

const Posts = ({ posts }) => {
  const postList = posts.map((post, i) => {
    return (
      <li key={i}>
        <PostItems key={post.id} to={ROUTES.POST_CONTAINER.linkTo(post.id)}>
          <p> {post.title}</p>
        </PostItems>
        <PostDes>{post.body}</PostDes>
      </li>
    );
  });

  return (
    <>
    <PostLists>
      {postList}
    </PostLists>
     </>
  );
};

Posts.propTypes =  {
  posts: PropTypes.array.isRequired
};

export default Posts;
