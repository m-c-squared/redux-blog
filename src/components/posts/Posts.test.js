import React from 'react';
import renderer from 'react-test-renderer';
import Posts from './Posts';
import { MemoryRouter } from 'react-router-dom';


describe('Posts', () => {
  it('matches a snapshot', () => {
    const posts = [
      {
        id:'1',
        title: 'asperiores ea ipsam voluptatibus modi minima quia sinta'
      },
      {
        id:'21',
        title: 'ullam ut quidem id aut vel consequunt' 
      }
    ];
    const tree = renderer.create(
      <MemoryRouter>
        <Posts posts={posts}/>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
