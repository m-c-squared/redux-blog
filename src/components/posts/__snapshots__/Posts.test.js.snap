// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts matches a snapshot 1`] = `
<ul
  className="sc-gZMcBi gIIBCr"
>
  <li>
    <a
      className="sc-gqjmRU xwxcz"
      href="/posts/1"
      onClick={[Function]}
    >
      <p>
         
        asperiores ea ipsam voluptatibus modi minima quia sinta
      </p>
    </a>
    <p
      className="sc-VigVT bblIYE"
    />
  </li>
  <li>
    <a
      className="sc-gqjmRU xwxcz"
      href="/posts/21"
      onClick={[Function]}
    >
      <p>
         
        ullam ut quidem id aut vel consequunt
      </p>
    </a>
    <p
      className="sc-VigVT bblIYE"
    />
  </li>
</ul>
`;
